Buchanan-Allison-LiDAR-Uncertainty-Class (1).py

```diff
@@ -0,0 +1,274 @@
+
+# # LiDAR Uncertainty: A Comparison of Two California Sites.
+
+# ## The Soap Site
+# ![image.png](attachment:image.png)
+# image credit: National Ecological Observation Network
+
+# In[13]:
+
+
+## The SJER Site
+get_ipython().system('[image.png](attachment:image.png)')
+image credit: National Ecological Observation Network
+
+
+# In[2]:
+
+
+# Import necessary packages
+import os
+import pathlib
+
+import earthpy as et
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import rasterstats as rs
+import rioxarray as rxr
+import seaborn as sns
+import xarray as xr
+
+# Download data from earth py
+et.data.get_data('spatial-vector-lidar')
+
+# General housekeeping
+home_dir = os.path.join(pathlib.Path.home(), 'earth-analytics',
+                        'data', 'spatial-vector-lidar')
+os.chdir(home_dir)
+
+# If the dir does not exist, create it
+output_path = os.path.join(home_dir,
+                           "outputs")
+
+
+# In[3]:
+
+
+class NEONDataLoader:
+
+    '''This class creates all of our necessary paths for downloads
+
+    parameters:
+    ----------
+    site_name = str
+    id_col = str
+    formatting_dict = key (?)
+
+    returns:
+    -------
+    Data is opened and transferred to correct locations.
+    '''
+
+    # Create necessary variables for SOAP and SJER sites.
+    base_dir_tmpl = os.path.join('california',
+                                 'neon-{site_name_low}-site')
+
+    insitu_path_tmpl = os.path.join('{base_dir}', '2013',
+                                    'insitu', 'veg{separator}structure',
+                                    'D17_2013_{site_name_up}_vegStr.csv')
+    plots_path_tmpl = os.path.join(
+        '{base_dir}', 'vector_data', '{site_name_up}{plot}_centroids.shp')
+    chm_path_tmpl = os.path.join(
+        '{base_dir}', '2013', 'lidar', '{site_name_up}_lidarCHM.tif')
+
+    # Create init function download both sets of data.
+    def __init__(self, site_name, id_col, formatting_dict={}):
+        self.site_name = site_name
+        self.id_col = id_col
+        self.formatting_dict = formatting_dict
+        self.formatting_dict['site_name_low'] = site_name.lower()
+        self.formatting_dict['site_name_up'] = site_name.upper()
+        self.formatting_dict['base_dir'] = self.base_dir_tmpl.format(
+            **self.formatting_dict)
+
+        self.insitu_path = self.insitu_path_tmpl.format(**self.formatting_dict)
+        self.plots_path = self.plots_path_tmpl.format(**self.formatting_dict)
+        self.chm_path = self.chm_path_tmpl.format(**self.formatting_dict)
+
+
+# In[4]:
+
+
+sjer_data_loader = NEONDataLoader('SJER', 'Plot_ID',
+                                  {'separator': '_', 'plot': '_plot'})
+sjer_data_loader
+
+
+# In[5]:
+
+
+soap_data_loader=NEONDataLoader('SOAP', 'ID', 
+                                {'separator':'-', 'plot':''})
+
+soap_data_loader.insitu_path
+
+
+# In[91]:
+
+
+class AnalyzeLidarUncertainty:
```

Comment:
I like how you divided this class into two, one to 'load' the data and one to 'run' the data. I think that could help with reproducibility with future data sets that require different steps to load, but once loaded could be run in the same way to pull out the max, means. Especially if you included some work here to standardize the column  and or file names?